shelley/chain-and-ledger/formal-spec/chain.tex

```diff
@@ -1382,9 +1360,14 @@ \subsection{Overlay Schedule}
       &
       \var{vkh}\leteq\hashKey{vk}
       \\
-      \bslot bhb \mapsto \var{gkh}\in\var{osched}
+      \var{slot} \leteq \bslot{bhb}
       &
+      \var{fSlot} \leteq \fun{firstSlot}~(\epoch{slot})
+      \\
       \var{gkh}\mapsto(\var{vkh},~\var{vrfh})\in\var{genDelegs}
+      \\
+      \isOverlaySlot{fSlot}{(\dom{genDelegs})}{slot}\\
+      \classifyOverlaySlot{fSlot}{(\dom{genDelegs})}{d}{\ActiveSlotCoeff}{slot} = \var{ghk}
```

Comment:
There is a bit of a disconnect between the formal spec and the code here, @nc6 . I'm not sure what the best thing to do is.

The formal spec has two separate rules, one for if the slot is an overlay lot and another rule otherwise. The code does a case analysis on a "double maybe" in order to capture both rules, and for this it uses a [helper function](https://github.com/input-output-hk/cardano-ledger-specs/blob/master/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Overlay.hs#L264) named `lookupInOverlaySchedule`.

In the code, `lookupInOverlaySchedule` is nice, in the spec it is less nice. I could change to code to _not_ use the helper, or I can leave this as I have it now...

Reply:
I think it's probably reasonable to leave as it is, given how simple that function is. We already have some cases where the spec and code diverge in terms of one vs multiple rules, so I don't think this is too egregious